onnxscript/evaluator.py

```diff
@@ -144,3 +204,99 @@ def call_ort(schema, args, kwargs, implicit_args=None):
     # Map ORT output values to the onnxscript representation-type.
     cast_result = [ort_to_os_value(x) for x in result]
     return cast_result[0] if len(cast_result) == 1 else cast_result
+
+
+def id(schema):
+    return schema.name, schema.domain, schema.since_version
+
+
+class ORTEvaluator(Evaluator):
+    """Evaluates ONNX ops using ONNX Runtime."""
+
+    def _eval(self, schema, inputs, attributes, closure):
+        return call_ort(schema, inputs, attributes, closure)
+
+
+ort_evaluator = ORTEvaluator()
+
+
+class ORTMixedEvaluator(ORTEvaluator):
+    """Evaluates ONNX ops using ONNX Runtime, unless an overriding python implementation
+    is registered. This is useful for higher-order ops such as Scan and SequenceMap,
+    allowing for python-based debugging.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._python_ops = {}
+
+    def use_graph_attribute(self, schema):
+        return id(schema) not in self._python_ops
+
+    def _eval(self, schema, inputs, attributes, closure):
+        if id(schema) in self._python_ops:
```

Comment:
You call `id(schema)` twice. You should store it in a variable.

Reply:
done